stl/inc/exception

```diff
@@ -21,10 +21,10 @@ _STL_DISABLE_CLANG_WARNINGS
 _STD_BEGIN
 
 #if _HAS_DEPRECATED_UNCAUGHT_EXCEPTION
-_EXPORT_STD extern "C++" _CXX17_DEPRECATE_UNCAUGHT_EXCEPTION _CRTIMP2_PURE bool __CLRCALL_PURE_OR_CDECL
+_EXPORT_STD extern "C++" _CXX17_DEPRECATE_UNCAUGHT_EXCEPTION _NODISCARD _CRTIMP2_PURE bool __CLRCALL_PURE_OR_CDECL

```

Comment:
Before I noticed your comment, it occurred to me that we should annotate the definitions consistently. I'll let you decide if that adds enough weight to do so in this PR.

Reply:
Now I can't find the precedent that I thought I found earlier. I'll apply the attributes (including the deprecations) to the definitions.